src/components/hooks/use-agentId-param.ts

```diff
@@ -0,0 +1,9 @@
+import { useParams } from 'next/navigation'
+
+export function useAgentIdParam() {
+  const params = useParams<{ agentId: string }>()
+  const agentId =
```

Comment:
instead of needing to force casting here; throw an error if the param is not found
```
if (!params?.agentId) {
   throw new Error('This hook can only be used in pages where there is a param called agentId');
}
```

Then you do not need to force the cast below

Reply:
Sounds good; that's fair